test-utils/utils/bucket_test.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"testing"
+)
+
+func TestExpandListURL(t *testing.T) {
+	table := []struct {
+		bucket    string
+		pathParts []interface{}
+		expect    string
+	}{
+		{
+			bucket:    "kubernetes-jenkins",
+			pathParts: []interface{}{"logs", "kubernetes-gce-e2e", 1458},
+			expect:    "https://www.googleapis.com/storage/v1/b/kubernetes-jenkins/o?prefix=%2Flogs%2Fkubernetes-gce-e2e%2F1458%2F",
```

Comment:
next time, maybe I will. Already got this working though :)

On Tue, Jun 28, 2016 at 3:13 PM, Minhan Xia <notifications@github.com>
wrote:

> In test-utils/utils/bucket_test.go
> <https://github.com/kubernetes/contrib/pull/1281#discussion_r68853952>:
>
> > +package utils
> > +
> > +import (
> > +	"testing"
> > +)
> > +
> > +func TestExpandListURL(t *testing.T) {
> > +	table := []struct {
> > +		bucket    string
> > +		pathParts []interface{}
> > +		expect    string
> > +	}{
> > +		{
> > +			bucket:    "kubernetes-jenkins",
> > +			pathParts: []interface{}{"logs", "kubernetes-gce-e2e", 1458},
> > +			expect:    "https://www.googleapis.com/storage/v1/b/kubernetes-jenkins/o?prefix=%2Flogs%2Fkubernetes-gce-e2e%2F1458%2F",
>
> It looks fine to me.
>
> I think I saw a golang gcs client somewhere. Why not use that one instead
> of writing a new one? I almost pull it last time when I was doing the gcs
> list stuff.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/contrib/pull/1281/files/cb8cb59b65946addb0eb0514c835879c0634316a#r68853952>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAnglvndl0tzYlJp9-CSm0xB3ShJh6uiks5qQZyEgaJpZM4JASOK>
> .
>